pkg/build/api/v1/types.go

```diff
@@ -402,6 +408,10 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// ExternalBuildStrategy defines input parameters specific to an External build.
+type ExternalBuildStrategy struct {
```

Comment:
> In previous conversations we were not creating os builds & were only going to update build config when external builds ran (think this was @smarterclayton's proposal). It does seem to make more sense to create os builds when triggering a job.

if you don't have a build per pipeline invocation instance, how are you going to track historical executions?

we can't just update the buildconfig every time we run a new pipeline.

and i don't like the idea of creating a new buildconfig each time the pipeline runs.  BuildConfigs represent intent, Builds represent execution, that's how the rest of the system works, so i don't see why we'd change it for this new strategy.